csrc/device_lower/pass/index.cpp

```diff
@@ -1581,14 +1581,31 @@ static inline DataType getMmaOutType(TensorView* mma_out) {
 }
 
 namespace {
+
 // Helper function to compute the index of the output of stmatrix (in shared
-// memory). We have two special cases (1) Handle tiles of size 8x8 and 16x8 and
-// (2) tiles of size 16x16. Each of these two cases has further two special
-// cases, (1) there is exactly one tile, (2) mulitple tiles thus multiple
-// stmatrix calls are issued in a for-loop. Please note that in a 16x16 tile,
-// the 4 8x8 tiles are stores/accessed in a m/column-major fasion. When we
-// mulitple tiles (whether it be 8x8, 18x6 or 16x16), the tiles are stored in a
-// m/column-major fashion.
+// memory). We currently handle the cases:
+// 1. stmatrix.x4 (we have 16x16 tiles)
+// 2. stmatrix.x2 (we have 16x8 tiles)
+// The assumption of the function is that we have 128-threads (warp group or 4
+// warps) The stmatrix call is inside a for-loop. We can think of 4-stmatrix (4
+// warps) calls being issued for every iteration of the for-loop. For example if
+// we are storing [64, 16] piece of memory using stmatrix.x2 (16x8), there will
+// be two iteration of the for-loop. In the first iteration we'll store [64 (0
+// ... 63), 8(0 ... 7)] and in the next iteration we'll store [64 (0 ... 63),
+// 8(8 ... 15)].
+
+// In each iteration, the 4-warps will be storing what we'll call
+// a warp group box. Each warp group box will have (at most) 4 tile boxes,
+// that's the memory stored by each stmatrix issued. This each tile-box is of
+// shape: [16, 16], [16, 8]; to be supported: [8, 8].
+
+// For example, if we are store [64, 16] using stmatrix.x2 [16, 8], then the
+// warp group box will be [64, 8] (there will be 2 warp group boxes). If we are
+// using stmatrix.x4 [16, 16] then the warp group box will be [64, 16].
+
+// To compute the index/offset into shared memory, we sum the offset of the warp
+// group box, the tile box inside the warp group box, and then finally the
+// offset of the thread inside the tile box.
 Val* hardCodedIndexGenerationForStMatrix(
```

Comment:
It would be nice to add a comment listing the final index you will get for all cases.